### PR TITLE
Remove warnings post move to scala 2.11

### PR DESCRIPTION
**Pull request by AnishShah**

### CODE REVIEW

**Comment by AnishShah** [comment]
Working on `SynchronizedBuffer` warning. Should I use `java.util.concurrent.ConcurrentLinkedQueue` as the Deprecation warning says ?

**Comment by szymonm** [comment]
I think so.

**Comment by AnishShah** [comment]
This removes all the warnings, except 

```
[warn] There may be incompatibilities among your library dependencies.
[warn] Here are some of the libraries that were evicted:
[warn] 	* com.google.guava:guava:13.0.1 -> 16.0.1
[warn] Run 'evicted' to see detailed eviction warnings
```

**Comment by pankajgupta** [comment]
Filed separate issue https://github.com/twitter/cassovary/issues/132 for the library eviction warnings. Perhaps @caniszczyk or @travisbrown have seen that elsewhere as well?

File: cassovary-core/src/main/scala/com/twitter/cassovary/graph/TestGraph.scala [hunk 1]
```diff
@@ -15,15 +15,16 @@ package com.twitter.cassovary.graph
...
+import scala.collection.JavaConverters._
 import scala.util.Random
-import scala.collection.mutable.ArrayBuffer
 
 /**
  * A simple implementation of a DirectedGraph
  */
 case class TestGraph(nodes: Node*) extends DirectedGraph {
-  val nodeTable = new mutable.HashMap[Int, Node]
+  val nodeTable = new HashMap[Int, Node]
```
Comment (pankajgupta) [inline cassovary-core/src/main/scala/com/twitter/cassovary/graph/TestGraph.scala]: it is considered best practice to explicitly specify mutable.HashMap and import only scala.collection.mutable (see http://docs.scala-lang.org/overviews/collections/overview.html starting from "A useful convention is...")

File: cassovary-core/src/main/scala/com/twitter/cassovary/graph/TestGraph.scala [hunk 2]
```diff
@@ -15,15 +15,16 @@ package com.twitter.cassovary.graph
 
 import com.twitter.cassovary.graph.StoredGraphDir._
 import com.twitter.cassovary.util.{Sampling, BinomialDistribution}
-import scala.collection.mutable
+import java.util.concurrent.ConcurrentLinkedQueue
+import scala.collection.mutable.HashMap
```
Comment (pankajgupta) [inline cassovary-core/src/main/scala/com/twitter/cassovary/graph/TestGraph.scala]: delete (see next comment)

**Comment by pankajgupta** [comment]
LGTM otherwise

**Pull request merged**